update allow_dependency_failure section with behavior for cancelled jobs (#2241)
buildkite/docs: pages/pipelines/dependencies.md

## Allowing dependency failures

> You can add the `allow_dependency_failure` attribute to any step that has dependencies. The step will then run when the depended-on jobs complete, fail or even did not run but not after a cancelation. If you cancel a job, any subsequent steps with `allow_dependency_failure: true` do not execute. However, even if you continue to run the next step, if there are any failures at all, the build will still end up failing.

Reviewer: I think this is a bit clearer:

```suggestion
You can add the `allow_dependency_failure` attribute to any step that has dependencies. The step will then run when the depended-on jobs complete, fail, or do not run. However, if you cancel a job, any subsequent steps with `allow_dependency_failure: true` do not execute. Note that even if you continue to run the next step, the build will still fail if there are any failures.
```

What do you think?
